src/mlpack/methods/reinforcement_learning/q_network.hpp

```diff
@@ -0,0 +1,59 @@
+/**
+ * @file q_network.hpp
+ * @author Nishant Kumar
+ *
+ * This file is consists of the q network.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_METHODS_RL_Q_NETWORK_HPP
+#define MLPACK_METHODS_RL_Q_NETWORK_HPP
+
+#include <mlpack/prereqs.hpp>
+#include "training_config.hpp"
+
+
+namespace mlpack {
+namespace rl {
+
+template <typename FeedForwardNetworkType>
+class QNetwork
```

Comment:
Form the example it looks like the `QNetwork` has a predefined layer structure?

Reply:
Yeah, that's what I meant.. although, is 
```
template <typename FeedForwardNetworkType = FFN<MeanSquaredError<>, GaussianInitialization>>
class QNetwork
``` 
valid? Sorry if it seems like a stupid question..

Reply:
Yes, looks valid to me, does this result in any error?

Reply:
```
error: ‘FFN’ does not name a type
 template <typename FeedForwardNetworkType = FFN<MeanSquaredError<>, GaussianInitialization>>
```
Thats the error I get